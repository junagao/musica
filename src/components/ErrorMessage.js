import React from 'react';
import PropTypes from 'prop-types';

const ErrorMessage = ({ error }) => (
  <div className="error-message">
    <p>The following unexpected error ocurred: {error}</p>
  </div>
);

ErrorMessage.propTypes = {
  error: PropTypes.string.isRequired,
};

export default ErrorMessage;
