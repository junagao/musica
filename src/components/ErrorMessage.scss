.error-message {
  color: #e53e3e;
  font-size: 1.25rem;
  padding: 2rem 0rem;
}
