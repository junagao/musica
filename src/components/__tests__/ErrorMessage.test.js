import React from 'react';
import { shallow } from 'enzyme';

import ErrorMessage from 'components/ErrorMessage';

describe('ErrorMessage', () => {
  let wrapper;
  const error = 'ERROR';

  beforeEach(() => {
    wrapper = shallow(<ErrorMessage error={error} />);
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one div element with className of error-message', () => {
    expect(wrapper.find('div').length).toEqual(1);
    expect(wrapper.find('div').hasClass('error-message')).toBeTruthy();
  });

  it('should render one p element with the error message', () => {
    expect(wrapper.find('p').at(0).length).toEqual(1);
    expect(
      wrapper
        .find('p')
        .at(0)
        .text(),
    ).toEqual(`The following unexpected error occurred: ${error}`);
  });
});
