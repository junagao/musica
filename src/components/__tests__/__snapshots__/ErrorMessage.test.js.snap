// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should render correctly given the required props 1`] = `
<div
  className="error-message"
>
  <p>
    The following unexpected error occurred: 
    ERROR
  </p>
</div>
`;
